### PR TITLE
FLW-2080: Logic for custom swipe action

### PR DESCRIPTION
**Pull request by DRadmir**

### CODE REVIEW

**Review by Viscount88 — APPROVED** [review]

File: fearless/Common/View/TableViewCell/SwipableTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+            let swipableVisibleCells = usedTableView?.visibleCells as? [SwipableTableViewCell] {
+            swipableVisibleCells.forEach { $0.closeSwipe() }
+            return true
+        }
+
+        return false
+    }
+}
+
+private enum SwipeDirectional {
```
Comment (bnsports) [inline fearless/Common/View/TableViewCell/SwipableTableViewCell.swift:462]: 'ended' looks like not direction, I think it should be named "SwipeState"

File: fearless/Common/View/TableViewCell/SwipableTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,474 @@
+import Foundation
+import UIKit
+import SwiftUI
+
+protocol SwipeButtonProtocol: UIButton {
+    var type: SwipeButtonType { get }
+}
+
+enum SwipeButtonType: Int, CaseIterable {
```
Comment (bnsports) [inline fearless/Common/View/TableViewCell/SwipableTableViewCell.swift:9]: rename please to SwipableCellButtonType. SwipeButtonType is very common, possible duplicates across the app or in external libs

File: fearless/Common/View/TableViewCell/SwipableTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+    case send
+    case receive
+    case teleport
+    case hide
+}
+
+protocol SwipeTableCellDelegate: AnyObject {
```
Comment (bnsports) [inline fearless/Common/View/TableViewCell/SwipableTableViewCell.swift:16]: different naming with cell class. please use the same

File: fearless/Common/View/TableViewCell/SwipableTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+    var openMenuTrigger: CGFloat {
+        actionButtonWidht * 0.7
+    }
+
+    var rightMenuButtons: [SwipeButtonProtocol] = []
+
+    var leftMenuButtons: [SwipeButtonProtocol] = []
+
+    /// Time for closing swipe action menu
+    var animationDuration: TimeInterval {
```
Comment (bnsports) [inline fearless/Common/View/TableViewCell/SwipableTableViewCell.swift:57]: you can name this variable "closeSwipeAnimationDuration" and use it without description comment. Don't thank me. I'm just doing my job.

**Review by bnsports — APPROVED** [review]

**Pull request merged**